Tools/CISupport/ews-build/steps.py

```diff
@@ -959,54 +1051,105 @@ def skip_build(self, reason):
 
     def start(self):
         patch_id = self.getProperty('patch_id', '')
-        if not patch_id:
-            self._addToLog('stdio', 'No patch_id found. Unable to proceed without patch_id.\n')
-            self.descriptionDone = 'No patch id found'
+        pr_number = self.getProperty('github.number', '')
+
+        if not patch_id and not pr_number:
+            self._addToLog('stdio', 'No patch_id or pr_number found. Unable to proceed without one of them.\n')
+            self.descriptionDone = 'No change found'
+            self.finished(FAILURE)
+            return None
+
+        if patch_id and pr_number:
+            self._addToLog('stdio', 'Both patch_id and pr_number found. Unable to proceed with both.\n')
+            self.descriptionDone = 'Error: both PR and patch number found'
             self.finished(FAILURE)
             return None
 
+        if patch_id and not self.validate_bugzilla(patch_id):
+            return None
+        if pr_number and not self.validate_github(pr_number):
+            return None
+
+        if self.verifyBugClosed:
+            self._addToLog('stdio', 'Bug is open.\n')
```

Comment:
Seems like we do not perform bug validation for PRs, so this statement would be misleading. We shouldn't print it in case of PR.